server/stream.go

```diff
@@ -2343,7 +2404,19 @@ func (mset *stream) setupMirrorConsumer() error {
 	// Filters
 	if mset.cfg.Mirror.FilterSubject != _EMPTY_ {
 		req.Config.FilterSubject = mset.cfg.Mirror.FilterSubject
+		if mset.cfg.Mirror.SubjectTransformDest != _EMPTY_ {
+			mirror.tr, _ = NewSubjectTransform(mset.cfg.Mirror.FilterSubject, mset.cfg.Mirror.SubjectTransformDest)
+		}
+	}
+
+	var filters []string
+	for _, tr := range mset.cfg.Mirror.SubjectTransforms {
+		// will not fail as already checked before that the transform will work
+		subjectTransform, _ := NewSubjectTransform(tr.Source, tr.Destination)
```

Comment:
I know you check for this error, and that right now it's fine, but I feel it's not safe against refactors and future changes that might alter the flow.

I think that price for single error check is pretty small to avoid possible future risk.

Reply:
This is because it's not obvious to me how to handle an error at that point. That's why it does check for an error earlier in the code path. If you check, although the signature of `setupMirrorConsumer()` returns and `error` the code actually always does `return nil` and the only couple of error conditions it handles are for potentially transient errors where it re-schedules re-trying to set the consumer later, while if `NewSubjectTransform` returns an error scheduling a retry later will not help because the transform destination is not going to suddenly fix itself. 

So I don't mind checking the error is, but what would you suggest as to what to do if it does return an error?

Reply:
Added the logging of an error if it happens

Reply:
Ok to resolve?